File: flutter_app/lib/screens/caregiver_setup_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:provider/provider.dart';
import '../providers/app_state.dart';
import '../models/caregiver.dart';
import '../theme/app_colors.dart';
import '../widgets/care_header.dart';
import '../widgets/care_card.dart';

class CaregiverSetupScreen extends StatefulWidget {
  const CaregiverSetupScreen({super.key});

  @override
  State<CaregiverSetupScreen> createState() => _CaregiverSetupScreenState();
}

class _CaregiverSetupScreenState extends State<CaregiverSetupScreen> {
  final _nameController = TextEditingController();
  final _relationController = TextEditingController();
  final _phoneController = TextEditingController();
  final _permissions = <String>{'appointments', 'reminders', 'help'};
  bool _nameValid = false;

  @override
  void initState() {
    super.initState();
    _nameController.addListener(_onNameChanged);
  }

  void _onNameChanged() {
    final valid = _nameController.text.trim().isNotEmpty;
    if (valid != _nameValid) {
      setState(() => _nameValid = valid);
    }
  }

  @override
  void dispose() {
    _nameController.dispose();
    _relationController.dispose();
    _phoneController.dispose();
    super.dispose();
  }

  void _addAndContinue() {
    if (_nameController.text.trim().isNotEmpty) {
      context.read<AppState>().addCaregiver(
        Caregiver(
          id: DateTime.now().millisecondsSinceEpoch.toString(),
          name: _nameController.text.trim(),
          relationship: _relationController.text.trim(),
          phone: _phoneController.text.trim(),
          permissions: _permissions.toList(),
        ),
      );
    }
    context.go('/confirmation');
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.pageBg,
      body: Column(
        children: [
          CareHeader(
            title: 'Add Help',
            onBack: () => context.canPop() ? context.pop() : context.go('/welcome'),
          ),
          Expanded(
            child: LayoutBuilder(
              builder: (context, constraints) {
                final isWide = constraints.maxWidth > 600;
                return SingleChildScrollView(
                  padding: const EdgeInsets.all(24),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      const Text(
                        'Add a trusted helper',
                        style: TextStyle(
                          fontSize: 28,
                          fontWeight: FontWeight.w900,
                          color: AppColors.heading,
                        ),
                      ),
                      const SizedBox(height: 8),
                      const Text(
                        'You can add a family member or friend to help manage your care.',
                        style: TextStyle(
                          fontSize: 18,
                          fontWeight: FontWeight.w500,
                          color: AppColors.mutedText,
                        ),
                      ),
                      const SizedBox(height: 24),
                      isWide
                          ? Row(
                              crossAxisAlignment: CrossAxisAlignment.start,
                              children: [
                                Expanded(child: _buildForm()),
                                const SizedBox(width: 24),
                                Expanded(child: _buildPermissions()),
                              ],
                            )
                          : Column(
                              children: [
                                _buildForm(),
                                const SizedBox(height: 24),
                                _buildPermissions(),
                              ],
                            ),
                      const SizedBox(height: 16),
                      Text(
                        'You can change permissions at any time in Settings.',
                        style: TextStyle(
                          fontSize: 18,
                          fontStyle: FontStyle.italic,
                          color: AppColors.mutedText,
                        ),
                      ),
                      const SizedBox(height: 24),
                    ],
                  ),
                );
              },
            ),
          ),
          _buildFooter(),
        ],
      ),
    );
  }

  Widget _buildForm() {
    return Column(
      children: [
        _inputField(
          controller: _nameController,
          label: "Helper's Full Name",
          hint: 'e.g. Sarah Jones',
          icon: Icons.person,
        ),
        const SizedBox(height: 16),
        _inputField(
          controller: _relationController,
          label: 'Relationship to You',
          hint: 'e.g. Daughter',
          icon: Icons.people,
        ),
        const SizedBox(height: 16),
        _inputField(
          controller: _phoneController,
          label: 'Phone Number',
          hint: '555-0123',
          icon: Icons.phone,
          keyboardType: TextInputType.phone,
        ),
      ],
    );
  }

  Widget _inputField({
    required TextEditingController controller,
    required String label,
    required String hint,
    required IconData icon,
    TextInputType keyboardType = TextInputType.text,
  }) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(
            fontSize: 18,
            fontWeight: FontWeight.w700,
            color: AppColors.heading,
          ),
        ),
        const SizedBox(height: 8),
        Container(
          decoration: BoxDecoration(
            color: AppColors.white,
            borderRadius: BorderRadius.circular(24),
            border: Border.all(color: AppColors.border, width: 3),
          ),
          child: TextField(
            controller: controller,
            keyboardType: keyboardType,
            style: const TextStyle(fontSize: 20, color: AppColors.heading),
            decoration: InputDecoration(
              hintText: hint,
              hintStyle: const TextStyle(color: AppColors.disabledText),
              prefixIcon: Padding(
                padding: const EdgeInsets.only(left: 16, right: 12),
                child: Icon(icon, color: AppColors.primaryAction, size: 24),
              ),
              prefixIconConstraints: const BoxConstraints(minWidth: 0),
              border: InputBorder.none,
              contentPadding: const EdgeInsets.symmetric(
                horizontal: 20, vertical: 18,
              ),
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildPermissions() {
    return CareCard(
      backgroundColor: AppColors.pageBg,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Container(
                padding: const EdgeInsets.all(10),
                decoration: BoxDecoration(
                  color: AppColors.blueBg,
                  borderRadius: BorderRadius.circular(14),
                ),
                child: const Icon(Icons.shield, color: AppColors.primaryAction, size: 24),
              ),
              const SizedBox(width: 12),
              const Expanded(
                child: Text(
                  'What they can see',
                  style: TextStyle(
                    fontSize: 20,
                    fontWeight: FontWeight.w700,
                    color: AppColors.heading,
                  ),
                ),
              ),
            ],
          ),
          const SizedBox(height: 16),
          _permissionTile(
            'appointments',
            'Appointments',
            'They can see when you have doctor visits.',
            Icons.calendar_today,
          ),
          const SizedBox(height: 12),
          _permissionTile(
            'reminders',
            'Reminders',
            'They can help you stay on track with medicine.',
            Icons.notifications,
          ),
          const SizedBox(height: 12),
          _permissionTile(
            'help',
            'Help Requests',
            'They will be notified if you ask for help.',
            Icons.help,
          ),
        ],
      ),
    );
  }

  Widget _permissionTile(String key, String title, String desc, IconData icon) {
    final isOn = _permissions.contains(key);
    return GestureDetector(
      onTap: () {
        setState(() {
          if (isOn) {
            _permissions.remove(key);
          } else {
            _permissions.add(key);
          }
        });
      },
      child: Container(
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: isOn ? AppColors.blueBg : AppColors.white,
          borderRadius: BorderRadius.circular(20),
          border: Border.all(
            color: isOn ? AppColors.primaryAction : AppColors.border,
            width: 3,
          ),
        ),
        child: Row(
          children: [
            Icon(icon, color: isOn ? AppColors.primaryAction : AppColors.mutedText, size: 24),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    title,
                    style: TextStyle(
                      fontSize: 18,
                      fontWeight: FontWeight.w700,
                      color: isOn ? AppColors.primaryAction : AppColors.heading,
                    ),
                  ),
                  Text(
                    desc,
                    style: const TextStyle(fontSize: 16, color: AppColors.mutedText),
                  ),
                ],
              ),
            ),
            if (isOn)
              const Icon(Icons.check_circle, color: AppColors.primaryAction, size: 24),
          ],
        ),
      ),
    );
  }

  Widget _buildFooter() {
    return Container(
      padding: const EdgeInsets.all(24),
      decoration: const BoxDecoration(
        color: AppColors.white,
        border: Border(top: BorderSide(color: AppColors.border, width: 4)),
      ),
      child: SafeArea(
        top: false,
        child: Row(
          children: [
            Expanded(
              child: SizedBox(
                height: 60,
                child: OutlinedButton(
                  onPressed: () => context.go('/confirmation'),
                  style: OutlinedButton.styleFrom(
                    foregroundColor: AppColors.heading,
                    side: const BorderSide(color: AppColors.border, width: 3),
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(20),
                    ),
                  ),
                  child: const Text(
                    'Skip for Now',
                    style: TextStyle(fontSize: 18, fontWeight: FontWeight.w700, height: 1.0),
                  ),
                ),
              ),
            ),
            const SizedBox(width: 16),
            Expanded(
              child: SizedBox(
                height: 60,
                child: ElevatedButton(
                  onPressed: _nameValid ? _addAndContinue : null,
                  style: ElevatedButton.styleFrom(
                    backgroundColor: AppColors.primaryAction,
                    foregroundColor: AppColors.white,
                    disabledBackgroundColor: AppColors.border,
                    disabledForegroundColor: AppColors.disabledText,
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(20),
                    ),
                  ),
                  child: const Text(
                    'Add Caregiver',
                    style: TextStyle(fontSize: 18, fontWeight: FontWeight.w700, height: 1.0),
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: react-native-app/src/screens/CaregiverSetupScreen.js
import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, ScrollView, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Ionicons } from '@expo/vector-icons';
import CareHeader from '../components/CareHeader';
import Colors from '../theme/colors';
import { useAppState } from '../context/AppContext';

const PERMISSION_OPTIONS = [
  { key: 'appointments', label: 'View Appointments', icon: 'calendar' },
  { key: 'reminders', label: 'Medication Reminders', icon: 'notifications' },
  { key: 'help', label: 'Respond to Help Requests', icon: 'hand-left' },
];

export default function CaregiverSetupScreen({ navigation }) {
  const { addCaregiver } = useAppState();
  const [name, setName] = useState('');
  const [relationship, setRelationship] = useState('');
  const [phone, setPhone] = useState('');
  const [permissions, setPermissions] = useState(new Set(['appointments', 'reminders', 'help']));

  const nameValid = name.trim().length > 0;

  const togglePermission = (key) => {
    const next = new Set(permissions);
    next.has(key) ? next.delete(key) : next.add(key);
    setPermissions(next);
  };

  const handleAdd = () => {
    addCaregiver({
      id: Date.now().toString(),
      name: name.trim(),
      relationship: relationship.trim() || 'Family',
      phone: phone.trim(),
      permissions: [...permissions],
    });
    navigation.navigate('Confirmation');
  };

  return (
    <SafeAreaView style={styles.safe}>
      <CareHeader
        title="Add Help"
        onBack={() => navigation.canGoBack() ? navigation.goBack() : navigation.navigate('Welcome')}
      />
      <ScrollView contentContainerStyle={styles.scroll}>
        <Text style={styles.heading}>Add a Caregiver</Text>
        <Text style={styles.body}>
          Add someone who helps take care of you. They can receive alerts and check on your schedule.
        </Text>

        <View style={styles.form}>
          <Text style={styles.label}>Full Name</Text>
          <TextInput
            style={styles.input}
            value={name}
            onChangeText={setName}
            placeholder="e.g., Sarah Johnson"
            placeholderTextColor={Colors.disabledText}
            accessibilityLabel="Full Name"
          />
          <Text style={styles.label}>Relationship</Text>
          <TextInput
            style={styles.input}
            value={relationship}
            onChangeText={setRelationship}
            placeholder="e.g., Daughter"
            placeholderTextColor={Colors.disabledText}
            accessibilityLabel="Relationship"
          />
          <Text style={styles.label}>Phone Number</Text>
          <TextInput
            style={styles.input}
            value={phone}
            onChangeText={setPhone}
            placeholder="e.g., 555-0101"
            placeholderTextColor={Colors.disabledText}
            keyboardType="phone-pad"
            accessibilityLabel="Phone Number"
          />
        </View>

        <View style={styles.permissionsSection}>
          <View style={styles.permissionsHeader}>
            <View style={styles.permIcon}>
              <Ionicons name="shield" size={24} color={Colors.primaryAction} />
            </View>
            <Text style={styles.permTitle}>What they can see</Text>
          </View>
          {PERMISSION_OPTIONS.map((p) => (
            <TouchableOpacity
              key={p.key}
              style={styles.permRow}
              onPress={() => togglePermission(p.key)}
              accessibilityRole="checkbox"
              accessibilityState={{ checked: permissions.has(p.key) }}
            >
              <Ionicons
                name={permissions.has(p.key) ? 'checkbox' : 'square-outline'}
                size={24}
                color={permissions.has(p.key) ? Colors.primaryAction : Colors.disabledText}
              />
              <Ionicons name={p.icon} size={20} color={Colors.mutedText} style={{ marginLeft: 12 }} />
              <Text style={styles.permLabel}>{p.label}</Text>
            </TouchableOpacity>
          ))}
        </View>

        <View style={styles.actions}>
          <TouchableOpacity
            style={[styles.primaryButton, !nameValid && styles.disabledButton]}
            onPress={handleAdd}
            disabled={!nameValid}
            accessibilityRole="button"
          >
            <Text style={[styles.primaryText, !nameValid && styles.disabledText]}>Add Caregiver</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.secondaryButton}
            onPress={() => navigation.navigate('Confirmation')}
          >
            <Text style={styles.secondaryText}>Skip for Now</Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  safe: { flex: 1, backgroundColor: Colors.pageBg },
  scroll: { padding: 24, paddingBottom: 48 },
  heading: { fontSize: 28, fontWeight: '900', color: Colors.heading, marginBottom: 8 },
  body: { fontSize: 18, color: Colors.mutedText, lineHeight: 28, marginBottom: 24 },
  form: { marginBottom: 24 },
  label: { fontSize: 16, fontWeight: '700', color: Colors.heading, marginBottom: 8, marginTop: 16 },
  input: {
    borderWidth: 3, borderColor: Colors.border, borderRadius: 16,
    paddingHorizontal: 16, paddingVertical: 14,
    fontSize: 18, color: Colors.heading, backgroundColor: Colors.white,
  },
  permissionsSection: { marginBottom: 32 },
  permissionsHeader: { flexDirection: 'row', alignItems: 'center', marginBottom: 16 },
  permIcon: {
    width: 44, height: 44, borderRadius: 14, backgroundColor: Colors.blueBg,
    justifyContent: 'center', alignItems: 'center', marginRight: 12,
  },
  permTitle: { fontSize: 20, fontWeight: '700', color: Colors.heading, flex: 1 },
  permRow: {
    flexDirection: 'row', alignItems: 'center',
    paddingVertical: 12, paddingHorizontal: 8,
  },
  permLabel: { fontSize: 18, fontWeight: '500', color: Colors.heading, marginLeft: 12, flex: 1 },
  actions: { gap: 12 },
  primaryButton: {
    backgroundColor: Colors.primaryAction, borderRadius: 20,
    paddingVertical: 18, alignItems: 'center',
  },
  primaryText: { fontSize: 20, fontWeight: '700', color: Colors.white },
  disabledButton: { backgroundColor: Colors.border },
  disabledText: { color: Colors.disabledText },
  secondaryButton: {
    backgroundColor: Colors.white, borderRadius: 20,
    paddingVertical: 18, alignItems: 'center',
    borderWidth: 3, borderColor: Colors.primaryAction,
  },
  secondaryText: { fontSize: 20, fontWeight: '700', color: Colors.primaryAction },
});
